### PR TITLE
Implement events to calculator

### PR DESCRIPTION
**Pull request by Thi-Ponce**
### Lesson Objectives:
- Handle events in a React app.
- Use React life cycle methods.
- Understand the mechanism of lifting state up.

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Thi-Ponce,

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
- The PR description and title looks professional 👍


## Required Changes ♻️
_Check the comments under the review._
- [x] Let's update the readme file and personalize it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/component/Calculator.jsx [hunk 1]
```diff
@@ -2,41 +2,56 @@ import React from 'react';
...
         <Container maxWidth="xs">
           <div className="calculator-container">
             <div className="result-container">
-              <h1 className="result">0</h1>
+              <h1 className="result">
+                {total || ''}
+                {operation || ''}
+                {next || ''}
+              </h1>
             </div>
```
Comment (Whoistolu) [inline src/component/Calculator.jsx:35]: - [x] [OPTIONAL] Well done implementing the operations of the calculator, however, you need to fix the style for the screen `result-container` class. Please ensure it aligns properly.
#### Your Design
Screenshot from 2022-03-17 23-43-42

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
```
Comment (Whoistolu) [inline src/logic/calculate.js]: - [x] [OPTIONAL] Inasmuch as this looks like a description it also looks like it is currently littering the code-base, kindly remove the unused code to have a clean code-base.

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    return {};
+  }
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
```
Comment (Whoistolu) [inline src/logic/calculate.js]: - [x] [OPTIONAL] Inasmuch as this looks like a description it also looks like it is currently littering the code-base, kindly remove the unused code to have a clean code-base.

File: src/logic/calculate.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+  // User pressed an operation after pressing '='
+  if (!obj.next && obj.total && !obj.operation) {
+    return { ...obj, operation: buttonName };
+  }
+
+  // User pressed an operation button and there is an existing operation
```
Comment (Whoistolu) [inline src/logic/calculate.js]: These short descriptions are fine 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Thi-Ponce 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**